singleobject/writer.go

```diff
@@ -0,0 +1,31 @@
+package singleobject
+
+import (
+	"bytes"
+	"github.com/actgardner/gogen-avro/schema/canonical"
+)
+
+type soWriter struct {
+	w      *bytes.Buffer
+	header []byte
+}
+
+func NewWriter(buf *bytes.Buffer, header []byte) *soWriter {
```

Comment:
Likewise this probably doesn't have to be a writer but just a convenience method that accepts an interface and encodes it to an io.Writer:

```

type Encodable interface {
	Serialize(io.Writer) error
	AvroUID() string
}

func Write(e Encodable, w io.Writer) (error) 
``` 

See the container package for a similar idea:
https://github.com/actgardner/gogen-avro/blob/master/container/writer.go#L94
https://github.com/actgardner/gogen-avro/blob/master/container/avro_record.go#L10-L13

Reply:
did the change code is much simpler